python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -44,6 +44,12 @@ class Value extends TObject {
         PointsToInternal::pointsTo(result, _, this, _)
     }
 
+    /** Gets the origin AST node for this value. */
+    AstNode getOrigin() {
```

Comment:
I think it makes more sense for this to return a `ControlFlowNode`, since the origin may depend on the splitting that has taken place.
Also, that way you can [simplify `hasLocationInfo`](https://github.com/Semmle/ql/blob/787b80f9ae66fc882eb7472bafd75d7d076fd490/python/ql/src/semmle/python/objects/ObjectAPI.qll#L89) using this predicate.

Reply:
Done!